docs/authentication.md

```diff
@@ -32,35 +34,48 @@ The Helios client will now need to present a client certificate to authenticate.
 This can be either a pre-issued X.509 certificate file and associated private
 key, or a certificate generated based on your SSH key.
 
-### Existing X.509 certificate
-
-This is actually not yet implemented.
+First we'll explain briefly how the ssh-agent signed certificates work, and then go through how to supply credentials for the `helios` CLI and the Helios API respectively.
 
 ### On-the-fly certificates with ssh-agent
 
 If you have an OpenSSH key and use [ssh-agent](http://linux.die.net/man/1/ssh-agent)
-(you probably do if you use `ssh`), Helios can generate an X509 certificate for
+(you probably do if you use `ssh`), Helios can generate an X.509 certificate for
 you on the fly. Here's how it works:
 
 1. The Helios client checks for a running SSH agent, and if one is found, asks
    for a list of your public SSH keys. Private keys are never exposed by
    ssh-agent.
 
-2. The Helios client generates a self-signed X.509 certificate containing your
-   SSH public key and username, which it sends to the HTTPS server.
+2. The Helios client generates a temporary X.509 certificate containing your username.
+   It then signs it with your private SSH key via ssh-agent and sends it to the HTTPS server.
 
-3. To prove that you own the corresponding private key, the Helios client asks
-   ssh-agent to cryptographically hash and sign part of the TLS handshake with
-   the server.
+3. The server looks up the public key for the username specified in the certificate, and
+   uses it to verify that the certificate was signed by the corresponding private key. If the
+   signature can be verified we know that the user is who they claim to be. Each user
+   essentially acts as a certificate authority.
```

Comment:
I think we should mention that this description of what the server does with the certificate is completely specific to the helios-master installation, i.e. however the admins have set it up themselves.